backend/src/modules/authentication/core/domain/UserAccount.ts

```diff
@@ -34,9 +41,37 @@ export function setPasswordForUserAccount(
   };
 }
 
-function onPasswordWasSet(state: UserAccount | undefined, event: PasswordWasSet): UserAccount {
+function onPasswordWasSet(state: UserAccount, event: PasswordWasSet): UserAccount {
   return new UserAccount({
-    email: event.email,
+    userId: event.userId,
+    email: state.email,
     password: event.password,
   });
 }
+
+export async function authenticateUser(
+  state: UserAccount | undefined,
+  command: { email: string; password: string },
```

Comment:
Chcemy moze zastosowac jakis value object dla Email / Password)? Zeby wiedziec, ze Email to rzeczywiscie Email, a Password np. ma minimum 5 znakow itp?

Reply:
done